ui/src/app/views/cohort-list/component.ts

```diff
@@ -71,12 +71,37 @@ export class CohortListComponent implements OnInit, OnDestroy {
   }
 
   public deleteCohort(cohort: Cohort): void {
+    if (this.awaitingReview || !this.writePermission) {
+      return;
+    }
     this.cohortsService.deleteCohort(this.wsNamespace, this.wsId, cohort.id).subscribe(() => {
       this.cohortList.splice(
         this.cohortList.indexOf(cohort), 1);
     });
   }
 
+  cloneCohort(cohort: Cohort): void {
+    if (this.awaitingReview || !this.writePermission) {
+      return;
+    }
```

Comment:
If we do need these conditions in each of the 4 functions in the component, can we consolidate this into a single function?

Reply:
Will do!